api/v1alpha1/aws_clusterconfig_types.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/variables"
+)
+
+//+kubebuilder:object:root=true
+
+// AWSClusterConfig is the Schema for the awsclusterconfigs API.
+type AWSClusterConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AWSClusterConfigSpec `json:"spec,omitempty"`
+}
+
+// AWSClusterConfigSpec defines the desired state of AWSClusterConfig.
+type AWSClusterConfigSpec struct {
+	// +optional
+	Region *Region `json:"region,omitempty"`
+}
+
+func (AWSClusterConfigSpec) VariableSchema() clusterv1.VariableSchema {
```

Comment:
Is this missing public docs?

Reply:
Yes - is it ok if i add them after I combine the generic and provider patch hooks though?

Reply:
I'm gonna merge and I will follow up with docs in follow up PR.